crypto/src/main/java/org/web3j/crypto/ECKeyPair.java

```diff
@@ -17,22 +17,34 @@
 import java.util.Arrays;
 
 import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.crypto.params.ECDomainParameters;
 import org.bouncycastle.crypto.params.ECPrivateKeyParameters;
 import org.bouncycastle.crypto.signers.ECDSASigner;
 import org.bouncycastle.crypto.signers.HMacDSAKCalculator;
 import org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey;
 import org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPublicKey;
+import org.bouncycastle.math.ec.ECCurve;
 
 import org.web3j.utils.Numeric;
 
-/** Elliptic Curve SECP-256k1 generated key pair. */
+/** Elliptic Curve SECP-256k1 or SECP-256r1 generated key pair. */
 public class ECKeyPair {
     private final BigInteger privateKey;
     private final BigInteger publicKey;
+    private final ECCurve ecCurve;
 
     public ECKeyPair(BigInteger privateKey, BigInteger publicKey) {
         this.privateKey = privateKey;
         this.publicKey = publicKey;
+        this.ecCurve =
+                ECDSASignature.K1_CURVE
```

Comment:
I made this change because we need to figure out what the Keypair curve is Sign#signMessage . Based on the curve i create the appropriate ECDSASignature